Transfer groundControl from staff to admin route

josh1248 opened the pull request:
### Description
Addresses #3031 by re-routing ground control from a staff to an admin route.
P.S.: still figuring out the backend. Will further changes be required?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update
- [ ] Code quality improvements

### How to test

- Enter the test environment as an admin, and see that you are able to enter ground control normally.
- Copy the ground control route (should be '/groundcontrol')
- Switch the test environment to be a staff
- Check that the ground control button is now missing.
- Paste the ground control route. A 404 route should be expected.

- Go to `src/commons/navigationBar/AcademyNavigationBar` and change the ground control route into a staff route.
- Log in as staff. Manually append '/groundcontrol' to the URL to get to the ground control screen.
- Attempt to delete an assessment, or mass publish an assessment through the ground control panel.
- A 403 Forbidden error should show up.
- Other functions, such as unpublishing or publishing of a single submission in the gradings table, should still be possible.

### Checklist

<!-- Please delete options that are not relevant. -->

- [X] I have tested this code - I have manually tested the website to check if I broke any frontend-backend links.
- [ ] I have updated the documentation

RichDom2185 reviewed (approved):
LGTM, thanks!

RichDom2185 commented:
Waiting for the BE PR before merging.

RichDom2185 commented on src/commons/sagas/RequestsSaga.ts:
```diff
@@ -250,13 +250,13 @@ export const getAchievements = async (tokens: Tokens): Promise<AchievementItem[]
 
 /**
- * GET /courses/{courseId}/admin/users/{studentCourseRegId}/goals
+ * GET /courses/{courseId}/staff/users/{studentCourseRegId}/goals
```
Let's not do this. `/admin` doesn't refer to the role; it refers to (course) administration.

It's a sub-resources belonging to the REST resource which is the course itself.

The role check is transparent (i.e. unknown to) the frontend; you shouldn't be able to tell based on the endpoint what are the checks.

josh1248 commented:
@RichDom2185 backend PR is ready now. I have replaced all non-dangerous routes to use a "/staff" instead of "/admin" URL to connect to the backend. I have also changed the routes in the backend directly. All seems good, but I am not sure if I have advertently broken some frontend-backend link. Any automated way to test it?

RichDom2185 commented:
> Any automated way to test it?

@josh1248 not really, I think best to just not touch it. Refer to my comments above. 

Basically your endpoints paths should be related to your resource. 

E.g. GET /users/{id}/name

or something like that

RichDom2185 reviewed (approved):
LGTM, thanks!
